src/cpp/library/cstdlib

```diff
@@ -108,11 +108,15 @@ long int strtol(const char * str, char ** endptr, int base);
 
 unsigned long int strtoul(const char * str, char ** endptr, int base);
 
-char * getenv(const char * name) {
-	size_t size = strlen(name) + 1;
-	char tmp[size];
-	strcpy(tmp, name);
-	return tmp;
+char *getenv(const char *name) {
+  if(name == NULL || *name == '\0')
+    return NULL; // Invalid input
+  size_t size = strlen(name) + 1;
+  char *tmp = (char *)malloc(sizeof(char) * size);
```

Comment:
Won't that trigger memleak checks? 

I guess the proper way is to use an infinite array or just use an alloca which doesn't add a dead STMT.